### PR TITLE
[12.0][ADD] helpdesk_mgmt_ticket_partner_restricted

### PR DESCRIPTION
**Pull request by ChristianSantamaria**
Adds partner restriction based on your company
It is implemented in the form view, it should be added to the search view soon

### CODE REVIEW

File: helpdesk_mgmt_ticket_partner_restricted/views/helpdesk_ticket.xml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+    <data>
+        <record id="helpdesk_ticket_form_restricted_partner_id" model="ir.ui.view">
+            <field name="name">helpdesk.ticket.form.restricted(partner_id)</field>
+            <field name="model">helpdesk.ticket</field>
+            <field name="inherit_id" ref="helpdesk_mgmt.ticket_view_form"/>
+            <field name="arch" type="xml">
+                <xpath expr="//field[@name='partner_id']" position="attributes">
+                    <attribute name="domain">['|',('company_id','=',company_id),('company_id','=',False)]</attribute>
```
Comment (dalonsod) [inline helpdesk_mgmt_ticket_partner_restricted/views/helpdesk_ticket.xml]: Could you test if this code also works? I'm not sure...
```suggestion
                    <attribute name="domain">[('company_id','in',[company_id, False])]</attribute>
```

Comment (ChristianSantamaria) [reply helpdesk_mgmt_ticket_partner_restricted/views/helpdesk_ticket.xml]: Yes it works, it is already updated

**Review by dalonsod — APPROVED** [review]
Locally tested, LGTM.

Anyway, please add this line to the `DESCRIPTION.rst` file:
`This addon can be used instead of disabling private addresses access, because this one can generate permission issues with the ticket user`

**Comment by dalonsod** [comment]
@lmiguens-solvos could you review? Thanks!

**Comment by lmiguens-solvos** [comment]
> @lmiguens-solvos could you review? Thanks!

I tried it. I created a new company in my environment, and I created a new contact in that company, and I have seen that indeed when creating a ticket in the new company it only showed me the contacts of this company, perfect.
But, when trying to save that ticket, it gives me an error, I don't know if it will only be a matter of my installation or not ... did you ever try to register a ticket?

This is the error:

Error:
Odoo Server Error

Traceback (most recent call last):
  File "C:\Users\Luisa\PycharmProjects\odoo-12\odoo\odoo\tools\cache.py", line 88, in lookup
    r = d[key]
  File "C:\Users\Luisa\PycharmProjects\odoo-12\odoo\odoo\tools\func.py", line 69, in wrapper
    return func(self, *args, **kwargs)
  File "C:\Users\Luisa\PycharmProjects\odoo-12\odoo\odoo\tools\lru.py", line 44, in __getitem__
    a = self.d[obj].me
KeyError: ('ir.model.data', <function IrModelData.xmlid_lookup at 0x05093C90>, 'helpdesk_mgmt.created_ticket_template')

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\Luisa\PycharmProjects\odoo-12\odoo\odoo\http.py", line 656, in _handle_exception
    return super(JsonRequest, self)._handle_exception(exception)
  File "C:\Users\Luisa\PycharmProjects\odoo-12\odoo\odoo\http.py", line 314, in _handle_exception
    raise pycompat.reraise(type(exception), exception, sys.exc_info()[2])
  File "C:\Users\Luisa\PycharmProjects\odoo-12\odoo\odoo\tools\pycompat.py", line 87, in reraise
    raise value
  File "C:\Users\Luisa\PycharmProjects\odoo-12\odoo\odoo\http.py", line 698, in dispatch
    result = self._call_function(**self.params)
  File "C:\Users\Luisa\PycharmProjects\odoo-12\odoo\odoo\http.py", line 346, in _call_function
    return checked_call(self.db, *args, **kwargs)
  File "C:\Users\Luisa\PycharmProjects\odoo-12\odoo\odoo\service\model.py", line 97, in wrapper
    return f(dbname, *args, **kwargs)
  File "C:\Users\Luisa\PycharmProjects\odoo-12\odoo\odoo\http.py", line 339, in checked_call
    result = self.endpoint(*a, **kw)
  File "C:\Users\Luisa\PycharmProjects\odoo-12\odoo\odoo\http.py", line 941, in __call__
    return self.method(*args, **kw)
  File "C:\Users\Luisa\PycharmProjects\odoo-12\odoo\odoo\http.py", line 519, in response_wrap
    response = f(*args, **kw)
  File "c:\users\luisa\pycharmprojects\odoo-12\odoo\addons\web\controllers\main.py", line 962, in call_kw
    return self._call_kw(model, method, args, kwargs)
  File "c:\users\luisa\pycharmprojects\odoo-12\odoo\addons\web\controllers\main.py", line 954, in _call_kw
    return call_kw(request.env[model], method, args, kwargs)
  File "C:\Users\Luisa\PycharmProjects\odoo-12\odoo\odoo\api.py", line 757, in call_kw
    return _call_kw_model_create(method, model, args, kwargs)
  File "C:\Users\Luisa\PycharmProjects\odoo-12\odoo\odoo\api.py", line 737, in _call_kw_model_create
    result = method(recs, *args, **kwargs)
  File "<decorator-gen-132>", line 2, in create
  File "C:\Users\Luisa\PycharmProjects\odoo-12\odoo\odoo\api.py", line 440, in _model_create_single
    return create(self, arg)
  File "c:\users\luisa\pycharmprojects\odoo-12\addons-helpdesk-slv\helpdesk\helpdesk_mgmt\models\helpdesk_ticket.py", line 176, in create
    res.send_partner_mail()
  File "c:\users\luisa\pycharmprojects\odoo-12\addons-helpdesk-slv\helpdesk\helpdesk_mgmt\models\helpdesk_ticket.py", line 90, in send_partner_mail
    self.env.ref('helpdesk_mgmt.created_ticket_template'). \
  File "C:\Users\Luisa\PycharmProjects\odoo-12\odoo\odoo\api.py", line 864, in ref
    return self['ir.model.data'].xmlid_to_object(xml_id, raise_if_not_found=raise_if_not_found)
  File "c:\users\luisa\pycharmprojects\odoo-12\odoo\odoo\addons\base\models\ir_model.py", line 1429, in xmlid_to_object
    t = self.xmlid_to_res_model_res_id(xmlid, raise_if_not_found)
  File "c:\users\luisa\pycharmprojects\odoo-12\odoo\odoo\addons\base\models\ir_model.py", line 1413, in xmlid_to_res_model_res_id
    return self.xmlid_lookup(xmlid)[1:3]
  File "<decorator-gen-25>", line 2, in xmlid_lookup
  File "C:\Users\Luisa\PycharmProjects\odoo-12\odoo\odoo\tools\cache.py", line 93, in lookup
    value = d[key] = self.method(*args, **kwargs)
  File "c:\users\luisa\pycharmprojects\odoo-12\odoo\odoo\addons\base\models\ir_model.py", line 1402, in xmlid_lookup
    raise ValueError('External ID not found in the system: %s' % xmlid)
ValueError: External ID not found in the system: helpdesk_mgmt.created_ticket_template

**Comment by dalonsod** [comment]
> raise ValueError('External ID not found in the system: %s' % xmlid)
> ValueError: External ID not found in the system: helpdesk_mgmt.created_ticket_template

That External ID is defined here: https://github.com/OCA/helpdesk/blob/12.0/helpdesk_mgmt/data/helpdesk_data.xml#L177

, and it seems to be created at this commit: https://github.com/OCA/helpdesk/commit/98e5ef7f4a2e074283da3674654f662a28f02eed#diff-bb2c820f8b13b25865ee77e6a83cdf6b1154736d1b77c1a571c5b36b6d5c72f3R177

So, it seems that you need to upgrade `helpdesk_mgmt` addon, or reinstall it.

**Comment by lmiguens-solvos** [comment]
> > raise ValueError('External ID not found in the system: %s' % xmlid)
> > ValueError: External ID not found in the system: helpdesk_mgmt.created_ticket_template
> 
> That External ID is defined here: https://github.com/OCA/helpdesk/blob/12.0/helpdesk_mgmt/data/helpdesk_data.xml#L177
> 
> , and it seems to be created at this commit: [OCA/helpdesk@98e5ef7#diff-bb2c820f8b13b25865ee77e6a83cdf6b1154736d1b77c1a571c5b36b6d5c72f3R177](https://github.com/OCA/helpdesk/commit/98e5ef7f4a2e074283da3674654f662a28f02eed#diff-bb2c820f8b13b25865ee77e6a83cdf6b1154736d1b77c1a571c5b36b6d5c72f3R177)
> 
> So, it seems that you need to upgrade `helpdesk_mgmt` addon, or reinstall it.

You are all right. I reinstalled `helpdesk_mgmt` addon and it works.

**Review by lmiguens-solvos — APPROVED** [review]
LGTM

**Pull request merged**